examples/conversion-gen/generators/conversion.go

```diff
@@ -0,0 +1,339 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package generators
+
+import (
+	"fmt"
+	"io"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"k8s.io/gengo/args"
+	conversiongenerator "k8s.io/gengo/examples/conversion-gen/generators/generator"
+	"k8s.io/gengo/generator"
+	"k8s.io/gengo/namer"
+	"k8s.io/gengo/types"
+
+	"k8s.io/klog"
+
+	conversionargs "k8s.io/code-generator/cmd/conversion-gen/args"
+)
+
+// These are the comment tags that carry parameters for conversion generation.
+const (
+	// e.g., "+k8s:conversion-gen=<peer-pkg>" in doc.go, where <peer-pkg> is the
+	// import path of the package the peer types are defined in.
+	// e.g., "+k8s:conversion-gen=false" in a type's comment will let
+	// conversion-gen skip that type.
+	tagName = "k8s:conversion-gen"
```

Comment:
No kubernetes specific stuff should go in this repo.

Reply:
Eh, I would tend to agree, but then I followed the pattern set by e.g. deepcopy or defaultergen, where there are a lot of k8s-specific stuff...

Reply:
Yeah, sigh, I am not sure how those got in here with k8s specific tags.

Reply:
Where is deepcopy kubernetes-dependent? Some test cases use Kubernetes strings. But generation is generic.

Reply:
here: https://github.com/kubernetes/gengo/blob/master/examples/deepcopy-gen/generators/deepcopy.go#L43

Reply:
@sttts : it's the same situation here, pretty much.

Reply:
ok, that's a string. That does not count :)

Doesn't conversion-gen use types like Scheme?

Reply:
Only for the example.

If that's a problem though, happy to make the example simpler, and move that logic back to code-generator.